dace/codegen/targets/sve/codegen.py

```diff
@@ -0,0 +1,588 @@
+        util.patch_ctypes()
+    res = old_generate_header(self, sdfg, global_stream, callsite_stream)
+    return res
+
+
+def generate_footer_hook(self, sdfg: SDFG, global_stream: CodeIOStream,
+                        callsite_stream: CodeIOStream):
+    global old_generate_footer
+    if contains_any_sve(sdfg):
+        util.patch_ctypes()
+    res = old_generate_footer(self, sdfg, global_stream, callsite_stream)
+    util.restore_ctypes()
+    return res
+
+
+@dace.registry.autoregister_params(name='sve')
+class SVEMap(TargetCodeGenerator):
+    target_name = 'armv8'
+    title = 'sve'
+    language = 'cpp'
+
+    def get_load_stride(self, sdfg: SDFG, state: SDFGState, node: nodes.Node,
+                        memlet: dace.Memlet) -> symbolic.SymExpr:
+        """Determines the stride of a load/store based on:
+            - The memlet subset
+            - The array strides
+            - The involved SVE loop stride"""
+
+        scope = util.get_sve_scope(sdfg, state, node)
+        if scope is None:
+            raise NotImplementedError('Not in an SVE scope')
+
+        sve_param = scope.map.params[-1]
+        sve_range = scope.map.range[-1]
+        sve_sym = dace.symbolic.symbol(sve_param)
+
+        array = sdfg.arrays[memlet.data]
+
+        # 1. Flatten the subset to a 1D-offset (using the array strides)
+        offset_1 = memlet.subset.at([0] * len(array.strides), array.strides)
+
+        if not offset_1.has(sve_sym):
+            raise util.NotSupportedError("SVE param does not occur in subset")
+
+        # 2. Replace the SVE loop param with its next (possibly strided) value
+        offset_2 = offset_1.subs(sve_sym, sve_sym + sve_range[2])
+
+        # 3. The load stride is the difference between both
+        stride = (offset_2 - offset_1).simplify()
+
+        return stride
+
+    def add_header(self, function_stream: CodeIOStream):
+        if self.has_generated_header:
+            return
+        self.has_generated_header = True
+
+        function_stream.write('#include <arm_sve.h>\n')
+
+        # TODO: Find this automatically at compile time
+        function_stream.write(f'#define {util.REGISTER_BYTE_SIZE} 64\n')
+
+    def __init__(self, frame_codegen: DaCeCodeGenerator, sdfg: dace.SDFG):
+        dace.SCOPEDEFAULT_SCHEDULE[
+            dace.ScheduleType.SVE_Map] = dace.ScheduleType.Sequential
+        dace.SCOPEDEFAULT_STORAGE[
+            dace.ScheduleType.SVE_Map] = dace.StorageType.CPU_Heap
+        self.has_generated_header = False
+
+        """
+        # Hook the generate_header in DaCeCodeGenerator
+        global old_generate_footer, old_generate_header
+        if old_generate_footer is None:
+            old_generate_header = DaCeCodeGenerator.generate_header
+            old_generate_footer = DaCeCodeGenerator.generate_footer
+        DaCeCodeGenerator.generate_header = generate_header_hook
+        DaCeCodeGenerator.generate_footer = generate_footer_hook
+        """
```

Comment:
Why is this commented out? Where does the hooking happen?

Reply:
There is no hooking happening anymore :)